packages/plugin-npm-cli/sources/commands/npm/login.ts

```diff
@@ -0,0 +1,162 @@
+
+  .command(`npm login [-r,--registry REGISTRY]`)
+  .categorize(`Npm-related commands`)
+  .describe(`login to registry`)
+
+  .detail(`
+    This command will ask you for your registry username, email, password, and One-time password (when it applies).
+  `)
+
+  .example(
+    `Login to the default registry`,
+    `yarn npm login`,
+  )
+
+  .example(
+    `Login to a custom registry`,
+    `yarn npm login --registry http://localhost:4873`,
+  )
+
+  .action(async ({cwd, stdin, stdout, registry}: {cwd: string, stdin: Readable, stdout: Writable, registry: string}) => {
+    const configuration = await Configuration.find(cwd, pluginConfiguration);
+
+    // @ts-ignore
+    const prompt = inquirer.createPromptModule({
+      input: stdin,
+      output: stdout,
+    });
+
+    const report = await StreamReport.start({configuration, stdout}, async report => {
+      const registryUrl = registry ? new URL(registry).toString() : null;
+      const credentials = await getCredentials(prompt);
+      const url = `/-/user/org.couchdb.user:${encodeURIComponent(credentials.name)}`;
+      const requestOptions = {
+        configuration,
+        ident: null,
+        json: true,
+        forceAuth: false,
+        registryUrl,
+      };
+
+      // First we try to login only using the credentials (no otp)
+      try {
+        const response = await npmHttpUtils.put(url, credentials, requestOptions);
+
+        // If login succeeds here, go ahead and save the token to the home config, report and exit
+        // @ts-ignore
+        await setAuthToken(configuration, response.token);
+
+        return report.reportInfo(MessageName.UNNAMED, SUCCESS_MESSAGE);
+      } catch(error) {
+        // If our first try is an error could be one of two options:
+        // - An actual error (such as invalid credentials)
+        // - Missing OTP
+        if (!authRequiresOtp(error))
+          return report.reportError(MessageName.AUTHENTICATION_INVALID, ERROR_MESSAGE);
+      }
+
+      // If missing OTP was the issue, ask the user for the OTP and try again
+      // setting the according otp header
+      const otp = await getOtp(prompt);
+
+      try {
+        const headers = getOtpHeaders(otp);
+        const response = await npmHttpUtils.put(url, credentials, {...requestOptions, headers});
+
+        // @ts-ignore
+        await setAuthToken(configuration, response.token);
+
+        return report.reportInfo(MessageName.UNNAMED, SUCCESS_MESSAGE);
+      } catch(error) {
+        return report.reportError(MessageName.AUTHENTICATION_INVALID, ERROR_MESSAGE);
+      }
+    });
+
+    return report.exitCode();
+  });
+
+async function setAuthToken(configuration: Configuration, npmAuthToken: string) {
+  const registry = npmConfigUtils.getRegistry(null, {configuration});
+
+  await Configuration.updateHomeConfiguration({
+    npmRegistries: {
+      [registry]: {
+        npmAuthToken,
+      },
+    },
```

Comment:
When the default registry is targeted, you can set the auth to the top-level of the configuration object.